notes/stv/20230125-red-deploy-01.txt

```diff
@@ -0,0 +1,952 @@
+                    --group \
+                    --owner \
+                    --stats \
+                    --progress \
+                    --exclude '~Trash' \
+                    --human-readable \
+                    --checksum \
+                    --recursive \
+                    --rsync-path 'sudo rsync' \
+                    "${sshuser:?}@${sshhost:?}:/home/fedora/zeppelin/notebook" \
+                    "${backname:?}"
+
+      		 rsync \
+                    --perms \
+                    --times \
+                    --group \
+                    --owner \
+                    --stats \
+                    --progress \
+                    --human-readable \
+                    --checksum \
+                    --recursive \
+                    --rsync-path 'sudo rsync' \
+                    "${sshuser:?}@${sshhost:?}:/home/fedora/zeppelin/conf/notebook-authorization.json" \
+                    "${backname:?}"
+
+                if [ -L 'latest' ]
+                then
+                    rm 'latest'
+                fi
+                ln -s "${backname:?}" 'latest'
+
+            popd
+        popd
+
+        ls -al /var/local/backups/notebooks/
+		
+		> 
+
+		total 48
+		drwxrwxr-x. 12 fedora fedora 4096 Jan 25 15:41 .
+		drwxr-xr-x.  5 fedora users  4096 Jul 27 10:41 ..
+		drwxrwxr-x.  3 fedora fedora 4096 Nov  2 06:31 20221102-050416-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Nov 22 08:34 20221122-083001-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Nov 23 08:19 20221123-081704-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Dec  4 14:12 20221204-141216-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 23 11:29 20230123-112907-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:34 20230125-153441-live-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:35 20230125-153517-edit-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:37 20230125-153750-edit-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:39 20230125-153909-live-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 25 15:40 20230125-154054-live-notebooks
+		lrwxrwxrwx.  1 fedora fedora   30 Jan 25 15:41 latest -> 20230125-154054-live-notebooks
+
+
+
+
+
+# -----------------------------------------------------
+# Make a copy of the latest notebooks.
+# Delete the public examples and spark-warehouse directories.
+#[user@data]
+
+        pushd /var/local/backups
+            pushd notebooks
+
+                datetime=$(date '+%Y%m%d-%H%M%S')
+                editname="${datetime:?}-edit-notebooks"
+
+                cp -r -L 'latest' "${editname}"
+
+                pushd "${editname}"
+
+                    rm -rf 'notebook/Public Examples'
+
+                    rm -rf 'notebook/spark-warehouse'
+
+                popd
+
+                rm 'latest'
+                ln -s "${editname}" 'latest'
+
+            popd
+        popd
```

Comment:
There is no reason to do this step. It looks like this was copied verbatim from my notes. At the time, there were old directories of notebooks that needed a specific step to delete them. These directories are no longer there, so this step is redundant.